### PR TITLE
TYP: Add type hints to the clib.loading.clib_full_names/load_libgmt functions

### PR DESCRIPTION
**Pull request by seisman**
Add type hints and simplify codes using the walrus operator.

### CODE REVIEW

File: pygmt/clib/loading.py [hunk 1]
```diff
@@ -96,20 +97,27 @@ def clib_names(os_name: str) -> list[str]:
     return libnames
 
 
-def clib_full_names(env=None):
+def clib_full_names(env: os._Environ | dict | None = None) -> Iterator[str]:
```
Comment (weiji14) [inline pygmt/clib/loading.py:100]: Not sure about using a private `os_Environ` here...

Comment (seisman) [reply pygmt/clib/loading.py]: `os.environ` returns a mapping object, so change it to `collections.abc.Mapping` in 3212ac94f3f79897a11cd90ff97a06323c752447

File: pygmt/clib/loading.py [hunk 2]
```diff
@@ -96,20 +97,27 @@ def clib_names(os_name: str) -> list[str]:
     return libnames
 
 
-def clib_full_names(env=None):
+def clib_full_names(env: Mapping | dict | None = None) -> Iterator[str]:
```
Comment (weiji14) [inline pygmt/clib/loading.py:100]: `collections.abc.Mapping` includes `dict`, see https://stackoverflow.com/questions/35690572/what-does-isinstance-with-a-dictionary-and-abc-mapping-from-collections-doing. Maybe we can remove the `dict` part?

```suggestion
def clib_full_names(env: Mapping | None = None) -> Iterator[str]:
```

**Review by weiji14 — APPROVED** [review]
One minor suggestion, pre-approving for now.

**Pull request merged**